fix(zone.js): make sure fakeasync use the same id pool with native

JiaLiPassion opened the pull request:
Close #54323

fakeAsync should use the same timerId pool with native, so they will not conflict.

jonathan-meier commented on packages/zone.js/lib/zone-spec/fake-async-test.ts:
```diff
@@ -61,12 +61,17 @@ const timers = {
   clearTimeout: global.clearTimeout,
-  clearInterval: global.clearInterval
+  clearInterval: global.clearInterval,
+  nativeSetTimeout: (global.setTimeout as any)[Zone.__symbol__('OriginalDelegate')],
+  nativeClearTimeout: (global.clearTimeout as any)[Zone.__symbol__('OriginalDelegate')],
```
Shouldn't these two lines rather be the following:
```
nativeSetTimeout: global[Zone.__symbol__('setTimeout')],
nativeClearTimeout: global[Zone.__symbol__('clearTimeout')],
```

If `setTimeout` or `clearTimeout` is patched again outside of `zone.js` but after the patching done by `zone.js`, the `OriginalDelegate` symbols will most likely be missing on these functions, so grabbing them from `global` seems safer.

atscott replied:
@JiaLiPassion Can you address this comment as well? I believe this is accurate and would be correct. The line where the unpatched methods are assigned directly to `global` is here: https://github.com/angular/angular/blob/ee3bb81f99474efe844b83065bd73496a24e2f22/packages/zone.js/lib/common/utils.ts#L403

alxhub commented:
@JiaLiPassion if you could rebase this, and I think the comment about multiple patches is accurate.

JiaLiPassion committed:
fix(zone.js): make sure fakeasync use the same id pool with native

Close #54323

fakeAsync should use the same timerId pool with native, so they will not
conflict.

JiaLiPassion commented:
@jonathan-meier 
thank you for the review and the comment, I have updated the PR.
@alxhub @atscott 
I have addressed the comment from @jonathan-meier , and also rebased the PR, please review.

atscott commented:
caretaker note: sync will require patching in these changes to the [3 failing tests in TGP](https://fusion2.corp.google.com/presubmit/619201803/OCL:619201803:BASE:619183137:1711509390806:7494ead2/targets) [cl/619480497](http://cl/619480497)

dylhunn commented:
This PR was merged into the repository by commit ddbf6bb038d101daf5280abbd2a0efaa0b7fd3a0.

jonathan-meier commented:
Thank you all for the help in landing this fix! It seems to work great so far 👍
